blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java

```diff
@@ -55,35 +66,102 @@ public BluePipeline create(Reachable parent) throws IOException {
                 }
             }
         }
-        TopLevelItem item = create(Jenkins.getInstance(), getName(), DESCRIPTOR, CustomOrganizationFolderDescriptor.class);
 
-        if (item instanceof OrganizationFolder) {
-            GitHubSCMNavigator gitHubSCMNavigator = new GitHubSCMNavigator(apiUrl, orgName, credentialId, credentialId);
-            if (sb.length() > 0) {
-                gitHubSCMNavigator.setPattern(sb.toString());
+        User authenticatedUser =  User.current();
+        if(authenticatedUser == null){
+            throw new ServiceException.UnauthorizedException("Must login to create a pipeline");
+        }
+
+        TopLevelItem item = null;
+        try {
+
+            item = create(Jenkins.getInstance(), getName(), DESCRIPTOR, CustomOrganizationFolderDescriptor.class);
+
+            if (item instanceof OrganizationFolder) {
+                if(credentialId != null) {
+                    validateCredentialId(credentialId, (AbstractFolder) item);
+
+                    //Find domain attached to this credentialId, if present check if it's BlueOcean specific domain then
+                    //add the properties otherwise simply use it
+                    Domain domain = CredentialsUtils.findDomain(credentialId, authenticatedUser);
+                    if(domain == null){ //this should not happen since validateCredentialId found the credential
+                        throw new ServiceException.BadRequestExpception(
+                                new ErrorMessage(400, "Failed to create pipeline")
+                                        .add(new ErrorMessage.Error("scm.credentialId",
+                                                ErrorMessage.Error.ErrorCodes.INVALID.toString(),
+                                                "No domain in user credentials found for credentialId: "+ scmConfig.getCredentialId())));
+                    }
+                    if(domain.test(new DomainRequirement())) {
+                        ((OrganizationFolder) item)
+                                .addProperty(
```

Comment:
No duplicate test needed, this code creates a new org folder and is add property right after creation.